src/cr/cube/matrix/assembler.py

```diff
@@ -784,6 +810,16 @@ def _cube_has_overlaps(self):
             and self._cube.valid_overlaps is not None
         )
 
+    def _dimension_derivations(self, dimension, order):
```

Comment:
can this be named `_derived_element_indices`, `_dimension_derivation` seems fishy to me